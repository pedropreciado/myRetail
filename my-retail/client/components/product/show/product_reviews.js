import React from 'react';
import StarRatings from 'react-star-ratings';

import ReviewBody from './product_review_body';
import dateParser from '../../../utils/date_parser';

export default class extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      showReviews: false
    };
    
    this.renderProCon = this.renderProCon.bind(this);
    this.renderReviews = this.renderReviews.bind(this);
    this.handleAllReviewsClick = this.handleAllReviewsClick.bind(this);
  }

  handleAllReviewsClick() {
    let showReviews = !this.state.showReviews;

    this.setState({ showReviews });
  }

  renderReviews() {
    let { reviews } = this.props; 

    return reviews.map((review) => {
      return (<ReviewBody review={review} />);
    });
  }
  
  renderProCon(reviewType) {
    let { 
      review, 
      datePosted,
      overallRating,
      screenName,
      title,
    } = this.props[reviewType];
    let { month, day, year } = dateParser(datePosted);

    return (
      <div className='pro-con-inner-container'>
        <div className='pro-con-header'>
          { reviewType.toUpperCase() }
          <a>{
            `most helpful 
          ${
            reviewType === 'pro' 
            ? '4-5' 
            : '1-2'
          } star review`
          }</a>
        </div>

        <div className='pro-con-body'>
          <div className='pro-con-body-header'>
          <StarRatings
          rating={Number(overallRating)} starRatedColor='red'
          numberOfStars={5} name='rating'
          starDimension='10px' starSpacing='2.5px'
          />
          <a>{ title }</a>
          </div>
          <p className='pro-con-review'>{ review }</p>
          <a className='pro-con-review author'>
          { `${screenName} ${month} ${day}, ${year}`}
          </a>
        </div>
      </ div>
    );
  }

  render() {
    let { rating, reviews } = this.props;
    let { showReviews } = this.state;
    
    return (
      <div id='ratings-container'>
        <div id='ratings-header'>
          <div>
          <StarRatings 
          rating={Number(rating)} starRatedColor='red' 
          numberOfStars={5} name='rating' 
          starDimension='20px' starSpacing='5px'
          /> overall
          </div>

          <div id='all-reviews-button' onClick={this.handleAllReviewsClick}>
          <a>{ `${showReviews ? 'hide' : 'show'} all ${reviews.length} reviews` }</a>
          </div>
        </div>
        { 
          showReviews
          ? (
            <div id='reviews-container'>
            { this.renderReviews() }
            </div>
          )
          : (
          <div id='pro-con-outer-container'>
            {this.renderProCon('pro')}
            {this.renderProCon('con')}
          </div>
          )
        }
      </div>
    );
  }
}