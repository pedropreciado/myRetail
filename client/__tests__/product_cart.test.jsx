import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ProductCart from '../components/product/show/product_cart';
import product_reviews from '../components/product/show/product_reviews';

configure({ adapter: new Adapter() });

let offers = {
  price: '1.99',
  qualifier: 'online only'
};
let promotions = ['free', 'just kidding'];

describe('ProductCart', () => {
  let productCart = shallow(
    <ProductCart promotions={promotions}
    offers={offers} code={0} 
    />
  );
  let priceContainer = productCart.find('#price-container');
  let increment = productCart.find('#plus');
  let decrement = productCart.find('#minus');

  it('renders correctly', () => {
    expect(productCart).toMatchSnapshot();
  });

  it('displays price', () => {
    expect(priceContainer.props().children[0].props.children).toBe('1.99');
  });

  it('displays qualifier', () => {
    expect(priceContainer.props().children[1].props.children).toBe('online only');
  });

  it('displays promotions', () => {
    let promo = productCart.find('.promo');
    
    expect(promo.exists()).toBe(true);
  });

  it('displays correct quantity amount on increment/decrement', () => {
    let initialAmount = productCart.state('quantity');
    
    increment.simulate('click');
    expect(productCart.state('quantity') === initialAmount).toBe(false);
    decrement.simulate('click');
    expect(productCart.state('quantity') === initialAmount).toBe(true);
  });

  it('displays 1 when decrement button is clicked at 1', () => {
    decrement.simulate('click');
    expect(productCart.state('quantity')).toBe(1);
  });

  it('displays both buttons when code is 0', () => {
    let buttonsContainer = productCart.find('#cart-buttons-container');
    
    expect(buttonsContainer.props().children.length).toBe(2);
  });
  
  it('displays only "ADD TO CART" button when code is 1', () => {
    let withCodeOne = shallow(
      <ProductCart promotions={promotions}
        offers={offers} code={1}
      />);
    let addToCart = withCodeOne.find('#add-to-cart');
    let pickUp = withCodeOne.find('#pick-up');
      
    expect(addToCart.exists()).toBe(true);
    expect(pickUp.exists()).toBe(false);
  });

  it('displays only "PICK UP ..." buttons when code is 2', () => {
    let withCodeTwo = shallow(
      <ProductCart promotions={promotions}
        offers={offers} code={2}
      />);
    let addToCart = withCodeTwo.find('#add-to-cart');
    let pickUp = withCodeTwo.find('#pick-up');

    expect(addToCart.exists()).toBe(false);
    expect(pickUp.exists()).toBe(true);
  });

  it('displays return information', () => {
    let returnsContainer = productCart.find('#returns-container');
    let text = returnsContainer.props().children[1];
    
    expect(typeof text.props.children).toBe('string');
  });
  
  it('displays cart options', () => {
    let cartOptionsContainer = productCart.find('#cart-options-container');

    expect(cartOptionsContainer.props(0).children.length).toBe(3);
  });
});
