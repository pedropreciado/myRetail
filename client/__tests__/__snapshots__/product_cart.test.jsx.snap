// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCart renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductCart
    code={0}
    offers={
        Object {
            "price": "1.99",
            "qualifier": "online only",
          }
    }
    promotions={
        Array [
            "free",
            "just kidding",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="price-container"
>
          <a>
                    1.99
          </a>
          <a
                    id="price-qualifier"
          >
                    online only
          </a>
</div>,
        <div
          id="promo-container"
>
          <div
                    className="promo"
          >
                    <FaTag
                              color="red"
                    />
                    <a>
                              free
                    </a>
          </div>
          <div
                    className="promo"
          >
                    <FaTag
                              color="red"
                    />
                    <a>
                              just kidding
                    </a>
          </div>
</div>,
        <div
          id="quantity-field"
>
          <a>
                    quantity: 
          </a>
          <div
                    id="quantity-button-container"
          >
                    <div
                              className="quantity-button"
                              id="minus"
                              onClick={[Function]}
                    >
                               - 
                    </div>
                    1
                    <div
                              className="quantity-button"
                              id="plus"
                              onClick={[Function]}
                    >
                               + 
                    </div>
          </div>
</div>,
        <div
          id="cart-buttons-container"
>
          <div
                    className="cart-button"
                    id="add-to-cart"
                    onClick={[Function]}
          >
                    ADD TO CART
          </div>
          <div
                    className="cart-button"
                    id="pick-up"
                    onClick={[Function]}
          >
                    PICK UP IN STORE
          </div>
</div>,
        <div
          id="returns-container"
>
          <div
                    id="returns-left"
          >
                    returns
          </div>
          <div
                    id="returns-right"
          >
                    This item must be returned within 30 days of the ship date. See return policy for details. Prices, promotions, styles an availability may vary by store and online.
          </div>
</div>,
        <div
          id="cart-options-container"
>
          <div
                    className="option-button"
          >
                    ADD TO REGISTRY
          </div>
          <div
                    className="option-button"
          >
                    ADD TO LIST
          </div>
          <div
                    className="option-button"
          >
                    SHARE
          </div>
</div>,
      ],
      "id": "cart-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a>
              1.99
</a>,
            <a
              id="price-qualifier"
>
              online only
</a>,
          ],
          "id": "price-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1.99",
            },
            "ref": null,
            "rendered": "1.99",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "online only",
              "id": "price-qualifier",
            },
            "ref": null,
            "rendered": "online only",
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="promo"
>
              <FaTag
                            color="red"
              />
              <a>
                            free
              </a>
</div>,
            <div
              className="promo"
>
              <FaTag
                            color="red"
              />
              <a>
                            just kidding
              </a>
</div>,
          ],
          "id": "promo-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FaTag
                  color="red"
/>,
                <a>
                  free
</a>,
              ],
              "className": "promo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "red",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "free",
                },
                "ref": null,
                "rendered": "free",
                "type": "a",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FaTag
                  color="red"
/>,
                <a>
                  just kidding
</a>,
              ],
              "className": "promo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "red",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "just kidding",
                },
                "ref": null,
                "rendered": "just kidding",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a>
              quantity: 
</a>,
            <div
              id="quantity-button-container"
>
              <div
                            className="quantity-button"
                            id="minus"
                            onClick={[Function]}
              >
                             - 
              </div>
              1
              <div
                            className="quantity-button"
                            id="plus"
                            onClick={[Function]}
              >
                             + 
              </div>
</div>,
          ],
          "id": "quantity-field",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "quantity: ",
            },
            "ref": null,
            "rendered": "quantity: ",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="quantity-button"
                  id="minus"
                  onClick={[Function]}
>
                   - 
</div>,
                1,
                <div
                  className="quantity-button"
                  id="plus"
                  onClick={[Function]}
>
                   + 
</div>,
              ],
              "id": "quantity-button-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " - ",
                  "className": "quantity-button",
                  "id": "minus",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": " - ",
                "type": "div",
              },
              1,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " + ",
                  "className": "quantity-button",
                  "id": "plus",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": " + ",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cart-button"
              id="add-to-cart"
              onClick={[Function]}
>
              ADD TO CART
</div>,
            <div
              className="cart-button"
              id="pick-up"
              onClick={[Function]}
>
              PICK UP IN STORE
</div>,
          ],
          "id": "cart-buttons-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "ADD TO CART",
              "className": "cart-button",
              "id": "add-to-cart",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "ADD TO CART",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "PICK UP IN STORE",
              "className": "cart-button",
              "id": "pick-up",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "PICK UP IN STORE",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="returns-left"
>
              returns
</div>,
            <div
              id="returns-right"
>
              This item must be returned within 30 days of the ship date. See return policy for details. Prices, promotions, styles an availability may vary by store and online.
</div>,
          ],
          "id": "returns-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "returns",
              "id": "returns-left",
            },
            "ref": null,
            "rendered": "returns",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This item must be returned within 30 days of the ship date. See return policy for details. Prices, promotions, styles an availability may vary by store and online.",
              "id": "returns-right",
            },
            "ref": null,
            "rendered": "This item must be returned within 30 days of the ship date. See return policy for details. Prices, promotions, styles an availability may vary by store and online.",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="option-button"
>
              ADD TO REGISTRY
</div>,
            <div
              className="option-button"
>
              ADD TO LIST
</div>,
            <div
              className="option-button"
>
              SHARE
</div>,
          ],
          "id": "cart-options-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ADD TO REGISTRY",
              "className": "option-button",
            },
            "ref": null,
            "rendered": "ADD TO REGISTRY",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ADD TO LIST",
              "className": "option-button",
            },
            "ref": null,
            "rendered": "ADD TO LIST",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SHARE",
              "className": "option-button",
            },
            "ref": null,
            "rendered": "SHARE",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="price-container"
>
            <a>
                        1.99
            </a>
            <a
                        id="price-qualifier"
            >
                        online only
            </a>
</div>,
          <div
            id="promo-container"
>
            <div
                        className="promo"
            >
                        <FaTag
                                    color="red"
                        />
                        <a>
                                    free
                        </a>
            </div>
            <div
                        className="promo"
            >
                        <FaTag
                                    color="red"
                        />
                        <a>
                                    just kidding
                        </a>
            </div>
</div>,
          <div
            id="quantity-field"
>
            <a>
                        quantity: 
            </a>
            <div
                        id="quantity-button-container"
            >
                        <div
                                    className="quantity-button"
                                    id="minus"
                                    onClick={[Function]}
                        >
                                     - 
                        </div>
                        1
                        <div
                                    className="quantity-button"
                                    id="plus"
                                    onClick={[Function]}
                        >
                                     + 
                        </div>
            </div>
</div>,
          <div
            id="cart-buttons-container"
>
            <div
                        className="cart-button"
                        id="add-to-cart"
                        onClick={[Function]}
            >
                        ADD TO CART
            </div>
            <div
                        className="cart-button"
                        id="pick-up"
                        onClick={[Function]}
            >
                        PICK UP IN STORE
            </div>
</div>,
          <div
            id="returns-container"
>
            <div
                        id="returns-left"
            >
                        returns
            </div>
            <div
                        id="returns-right"
            >
                        This item must be returned within 30 days of the ship date. See return policy for details. Prices, promotions, styles an availability may vary by store and online.
            </div>
</div>,
          <div
            id="cart-options-container"
>
            <div
                        className="option-button"
            >
                        ADD TO REGISTRY
            </div>
            <div
                        className="option-button"
            >
                        ADD TO LIST
            </div>
            <div
                        className="option-button"
            >
                        SHARE
            </div>
</div>,
        ],
        "id": "cart-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a>
                1.99
</a>,
              <a
                id="price-qualifier"
>
                online only
</a>,
            ],
            "id": "price-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1.99",
              },
              "ref": null,
              "rendered": "1.99",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "online only",
                "id": "price-qualifier",
              },
              "ref": null,
              "rendered": "online only",
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="promo"
>
                <FaTag
                                color="red"
                />
                <a>
                                free
                </a>
</div>,
              <div
                className="promo"
>
                <FaTag
                                color="red"
                />
                <a>
                                just kidding
                </a>
</div>,
            ],
            "id": "promo-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FaTag
                    color="red"
/>,
                  <a>
                    free
</a>,
                ],
                "className": "promo",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": "red",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "free",
                  },
                  "ref": null,
                  "rendered": "free",
                  "type": "a",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FaTag
                    color="red"
/>,
                  <a>
                    just kidding
</a>,
                ],
                "className": "promo",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": "red",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "just kidding",
                  },
                  "ref": null,
                  "rendered": "just kidding",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a>
                quantity: 
</a>,
              <div
                id="quantity-button-container"
>
                <div
                                className="quantity-button"
                                id="minus"
                                onClick={[Function]}
                >
                                 - 
                </div>
                1
                <div
                                className="quantity-button"
                                id="plus"
                                onClick={[Function]}
                >
                                 + 
                </div>
</div>,
            ],
            "id": "quantity-field",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "quantity: ",
              },
              "ref": null,
              "rendered": "quantity: ",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="quantity-button"
                    id="minus"
                    onClick={[Function]}
>
                     - 
</div>,
                  1,
                  <div
                    className="quantity-button"
                    id="plus"
                    onClick={[Function]}
>
                     + 
</div>,
                ],
                "id": "quantity-button-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " - ",
                    "className": "quantity-button",
                    "id": "minus",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": " - ",
                  "type": "div",
                },
                1,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " + ",
                    "className": "quantity-button",
                    "id": "plus",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": " + ",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="cart-button"
                id="add-to-cart"
                onClick={[Function]}
>
                ADD TO CART
</div>,
              <div
                className="cart-button"
                id="pick-up"
                onClick={[Function]}
>
                PICK UP IN STORE
</div>,
            ],
            "id": "cart-buttons-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "ADD TO CART",
                "className": "cart-button",
                "id": "add-to-cart",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "ADD TO CART",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "PICK UP IN STORE",
                "className": "cart-button",
                "id": "pick-up",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "PICK UP IN STORE",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="returns-left"
>
                returns
</div>,
              <div
                id="returns-right"
>
                This item must be returned within 30 days of the ship date. See return policy for details. Prices, promotions, styles an availability may vary by store and online.
</div>,
            ],
            "id": "returns-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "returns",
                "id": "returns-left",
              },
              "ref": null,
              "rendered": "returns",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This item must be returned within 30 days of the ship date. See return policy for details. Prices, promotions, styles an availability may vary by store and online.",
                "id": "returns-right",
              },
              "ref": null,
              "rendered": "This item must be returned within 30 days of the ship date. See return policy for details. Prices, promotions, styles an availability may vary by store and online.",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="option-button"
>
                ADD TO REGISTRY
</div>,
              <div
                className="option-button"
>
                ADD TO LIST
</div>,
              <div
                className="option-button"
>
                SHARE
</div>,
            ],
            "id": "cart-options-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ADD TO REGISTRY",
                "className": "option-button",
              },
              "ref": null,
              "rendered": "ADD TO REGISTRY",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ADD TO LIST",
                "className": "option-button",
              },
              "ref": null,
              "rendered": "ADD TO LIST",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SHARE",
                "className": "option-button",
              },
              "ref": null,
              "rendered": "SHARE",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
